### ISSUE TITLE
[Bug] System.IdentityModel.Tokens.Jwt 7.* breaks oidc configuration parsing for common providers like Google

### ISSUE THREAD

**Issue by segfault** [AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet#2347]
**Which version of Microsoft.IdentityModel are you using?**
System.IdentityModel.Tokens.Jwt 7.0.0+ including the current 7.0.2

**Where is the issue?**
 * [ ] M.IM.JsonWebTokens
 * [ ] M.IM.KeyVaultExtensions
 * [ ] M.IM.Logging
 * [ ] M.IM.ManagedKeyVaultSecurityKey
 * [ ] M.IM.Protocols
 * [X] M.IM.Protocols.OpenIdConnect
 * [ ] M.IM.Protocols.SignedHttpRequest
 * [ ] M.IM.Protocols.WsFederation
 * [ ] M.IM.TestExtensions
 * [ ] M.IM.Tokens
 * [ ] M.IM.Tokens.Saml
 * [ ] M.IM.Validators
 * [ ] M.IM.Xml
 * [X] S.IM.Tokens.Jwt
 * Other (please describe)

**Is this a new or an existing app?**
The app is in production and I have upgraded to a new version of Microsoft.IdentityModel.*

**Repro**
Use AddOpenIdConnect to configure the metadata url to use Google's endpoint: `https://accounts.google.com/.well-known/openid-configuration`

**Expected behavior**
The metadata successfully parsed and functional oidc auth.

**Actual behavior**
Auth broken due to an InvalidOperationException (IDX20803) due to a failure to parse the metadata. Reported as "Unable to obtain configuration"

**Possible solution**
Offhand I presume this is related to the stricter parsing noted in #2325,#2336 and the switch in json parser #2224

**Additional context / logs / screenshots / links to code**
```
System.InvalidOperationException: IDX20803: Unable to obtain configuration from: 'https://accounts.google.com/.well-known/openid-configuration'.
   at Microsoft.IdentityModel.Protocols.ConfigurationManager`1.GetConfigurationAsync(CancellationToken cancel)
   at Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler.HandleChallengeAsyncInternal(AuthenticationProperties properties)
   at Microsoft.AspNetCore.Authentication.OpenIdConnect.OpenIdConnectHandler.HandleChallengeAsync(AuthenticationProperties properties)
   at Microsoft.AspNetCore.Authentication.AuthenticationHandler`1.ChallengeAsync(AuthenticationProperties properties)
   at Microsoft.AspNetCore.Authentication.AuthenticationService.ChallengeAsync(HttpContext context, String scheme, AuthenticationProperties properties)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.<>c__DisplayClass0_0.<<HandleAsync>g__Handle|0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddlewareImpl.<Invoke>g__Awaited|8_0(ExceptionHandlerMiddlewareImpl middleware, HttpContext context, Task task)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddlewareImpl.HandleException(HttpContext context, ExceptionDispatchInfo edi)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddlewareImpl.<Invoke>g__Awaited|8_0(ExceptionHandlerMiddlewareImpl middleware, HttpContext context, Task task)
   at Serilog.AspNetCore.RequestLoggingMiddleware.Invoke(HttpContext httpContext)
   at Sentry.AspNetCore.SentryMiddleware.InvokeAsync(HttpContext context, RequestDelegate next)
   at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass6_1.<<UseMiddlewareInterface>b__1>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
```

**Comment by brentschmaltz** [comment]
@segfault I added some tests [here](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/8085b721fb2d4f1ebde31c5769d8c690c6382256/test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/ConfigurationManagerTests.cs#L30).
We are able to obtain the OIDC metadata from google.

Are you using IdentityModel 7.x with a released version of Asp.net (7.x or lower)?

**Comment by segfault** [comment]
Running against the latest 7.x on Linux and MacOS. I'll try to run that test locally later today. Downgrading to the most recent 6.x identity model release immediately fixes the issue.

Example non-functional package reference list from the csproj:
```

  <ItemGroup>
    <PackageReference Include="MessagePack" Version="2.5.129" />
    <PackageReference Include="MessagePack.AspNetCoreMvcFormatter" Version="2.5.129" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Google" Version="7.0.11" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="7.0.11" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.OpenIdConnect" Version="7.0.11" />
    <PackageReference Include="Microsoft.AspNetCore.SpaServices.Extensions" Version="7.0.11" />
    <PackageReference Include="Microsoft.AspNetCore.Identity" Version="2.2.0" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="7.0.11" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.UI" Version="7.0.11" />
    <PackageReference Include="Microsoft.AspNetCore.DataProtection.StackExchangeRedis" Version="7.0.11" />
    <PackageReference Include="Microsoft.Extensions.Identity.Stores" Version="7.0.11" />
    <PackageReference Include="Microsoft.Extensions.Identity.Core" Version="7.0.11" />
    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="7.0.11" />
    <PackageReference Include="OpenTelemetry" Version="1.6.0" />
    <PackageReference Include="Sentry" Version="3.39.1" />
    <PackageReference Include="Sentry.AspNetCore" Version="3.39.1" />
    <PackageReference Include="Sentry.Serilog" Version="3.39.1" />
    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="7.0.2" />
    <PackageReference Include="prometheus-net" Version="8.0.1" />
    <PackageReference Include="prometheus-net.AspNetCore" Version="8.0.1" />
    <PackageReference Include="prometheus-net.AspNetCore.HealthChecks" Version="8.0.1" />
    <PackageReference Include="prometheus-net.DotNetRuntime" Version="4.4.0" />
    <PackageReference Include="Serilog" Version="3.0.1" />
    <PackageReference Include="Serilog.AspNetCore" Version="7.0.0" />
    <PackageReference Include="Serilog.Enrichers.Environment" Version="2.2.0" />
    <PackageReference Include="Serilog.Enrichers.Process" Version="2.0.2" />
    <PackageReference Include="Serilog.Enrichers.Thread" Version="3.1.0" />
    <PackageReference Include="Serilog.Expressions" Version="3.4.1" />
    <PackageReference Include="Serilog.Extensions.Hosting" Version="7.0.0" />
    <PackageReference Include="Serilog.Extensions.Logging" Version="7.0.0" />
    <PackageReference Include="Serilog.Settings.Configuration" Version="7.0.1" />
    <PackageReference Include="Serilog.Sinks.Async" Version="1.5.0" />
    <PackageReference Include="Serilog.Sinks.Console" Version="4.1.0" />
    <PackageReference Include="Serilog.Sinks.Debug" Version="2.0.0" />
    <PackageReference Include="Serilog.Sinks.File" Version="5.0.0" />
    <PackageReference Include="Serilog.Formatting.Compact" Version="1.1.0" />
    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.5.0" />
    <PackageReference Include="NetDevPack.Security.Jwt" Version="5.0.17" />
    <PackageReference Include="NetDevPack.Security.Jwt.AspNetCore" Version="7.1.3" />
    <PackageReference Include="NetDevPack.Security.JwtExtensions " Version="7.0.0" />
    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
    <PackageReference Include="System.CommandLine" Version="2.0.0-beta4.22272.1" />
    <PackageReference Include="System.CommandLine.NamingConventionBinder" Version="2.0.0-beta4.22272.1" />
    <PackageReference Include="System.CommandLine.Rendering" Version="0.4.0-alpha.22272.1" />
    <PackageReference Include="TimeZoneConverter" Version="6.1.0" />
    <PackageReference Include="Microsoft.Extensions.Caching.StackExchangeRedis" Version="7.0.11" />
    <PackageReference Include="Microsoft.SourceLink.Common" Version="1.1.1">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.SourceLink.GitLab" Version="1.1.1">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>
```

The dotnet runtime associated with the stacktrace:
```
Microsoft.AspNetCore.App 7.0.11 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 7.0.11 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

Tested on MacOS 13.6 and Ubuntu 22.04 `5.15.0-83-generic`

**Comment by brentschmaltz** [comment]
@segfault thanks for the quick response.
We have not thought about asp.net 7 and IdentityModel 7.
We have tested IdentityModel 7 with asp.net 8.

**Comment by senzacionale** [comment]
Same problem for me. IdentityModel 7 and .net 7.

Thank you for adding this to next Milestone

**Comment by keegan-caruso** [comment]
Doesn't appear to be about the aspnet core 7 x Identity Model 7 combination. Looks to be more related to the parsing and the google doc in IM7 regardless of aspnet core version.

**Comment by segfault** [comment]
It was for me. IM7 failed with the previous version of system.text.json
No issues when I switched to dotnet 8 and IM7

**Comment by keegan-caruso** [comment]
interesting, could you share a minimal repro? I make a new net7 Web Api with Jwt Bearer and updated the Identity Model assemblies to 7.1.2 and I don't see an error. If you can provide the repro I'm happy to dig into it.

**Comment by segfault** [comment]
It may well be fixed in `7.1.2`, when I encountered the problem `7.0.2` was the current release. When I get a chance I'll pull the env from that time and see if using `7.1.2` works out of the box.

**Comment by brentschmaltz** [comment]
@segfault @senzacionale This has been fixed with this PR: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2491

We hope to have a release out soon.
